include/lsst/meas/astrom/pessimisticPatternMatcherUtils.h

```diff
@@ -23,26 +23,143 @@
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
 
+#include <Eigen/Dense>
 #include <ndarray.h>
 
 namespace lsst {
 namespace meas {
 namespace astrom {
+
+struct PatternResult {
+    std::vector<std::pair<uint16_t, uint16_t>> candidate_pairs;
+    std::vector<double> shift_rot_matrix;
+    double cos_shift;
+    double sin_rot;
+    bool success = false;
+};
+
+struct RotationTestResult {
+    double cos_rot_sq;
+    Eigen::Vector3d proj_ref_ctr_delta;
+    Eigen::Matrix<double, 3, 3> shift_matrix;
```

Comment:
All references to `Eigen::Matrix<double, 3, 3>` [can be more concisely written](https://eigen.tuxfamily.org/dox/group__matrixtypedefs.html) as `Eigen::Matrix3d`.